src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+  if (state === 'poweredOn') {
+    beacon.advertiseUrl("https://goo.gl/9FomQC", {name: 'Edison'});
+  }   
+}
+{% endhighlight %}
+
+That really couldn't be much easier. Below you see that Chrome finds
+my device nicely.
+
+> NOTE: The Physical Web enabled devices only show up on your phone
+> when Bluetooth is turned on, and you launch Chrome (currently only
+> works with Beta). Additionally, you have to opt into the feature
+> first time you launch Chrome beta.
+
+<img alt="Chrome announces nearby Physical Web beacons" src="/web/updates/images/2016/03/web-enabled-internet-of-things/nearby.png" style="width: 50%; float: left"/>
+<img alt="The web app URL is listed" src="/web/updates/images/2016/03/web-enabled-internet-of-things/physicalweb.png" style="width: 50%; float: left"/>
+
+## Communicating with the sensor/actuator
+
+For talking to my board enhancements I used Johnny-Five. In simple
+cases like this, this is not strictly easier than talking manually,
+but for bigger projects it can be a real help. Johnny-Five has a nice
+abstraction for talking to a TMP36 sensor, but for some reason it only
+returned "undefined" as the current temperature so I went ahead and
+read the temperature value manually.
+
+Below you find the simple code for listening to temperature changes
+as well as setting the initial LED color.
+
+{% highlight javascript %}
+var five = require("johnny-five");
+var Edison = require("edison-io");
+var board = new five.Board({
+  io: new Edison()
+});
+
+board.on("ready", function() {
+  var led = new five.Led.RGB({
+      pins: {
+          red: 3,
+          green: 5,
+          blue: 6
+      },
+  });
+
+  colorCharacteristic._led = led;
+  led.color(colorCharacteristic._value);
+  led.intensity(30);
+
+  board.analogRead("A0", function(raw) {
+    var mV = 5 * 1000 * (raw / 1024);
+    var value = (mV / 10) - 50;
+    temperatureCharacteristic.valueChange(value);
+  });
+}
+{% endhighlight %}
+
+Don't worry about the `*Characteristic` variables above. These are
+what I use for communicating via Bluetooth. As you notice, I talk to
+the TMP36 via the analog A0 port and the voltage legs on the color
+LED cathode are connected to digital pin 3, 5 and 6, which happens
+to be PWM (pulse-wide modulation) pins on the Edison Arduino breakout
+board.
+
+![Edison board](/web/updates/images/2016/03/web-enabled-internet-of-things/board.jpg)
+
+## Talking Bluetooth
+
+Talking Bluetooth couldn't be much easier than it is with noble.
+
+In the following example, we create two characteristics, one for the
+LED and one for the temperature sensor. The former allows reading
+the current LED color as well as setting a new color. The latter
+allows subscribing to temperature change events.
+
+> NOTE: Initially I had some problems with the Bluetooth connection
+> being a bit unstable and not working on startup or bailing out during
+> connecting with a Frame Reassemble failure.
+>
+> If that happens, typing `rfkill block bluetooth` followed by
```

Comment:
Maybe this should be a nice hint somewhere else and a bit bolder- don't use USB power as it might not be enough - use a proper power supply etc